make-archive.pl

```diff
@@ -19,13 +19,18 @@
 
 sub main {
     GetOptions(
-        'executable-name=s' => \my $executable_name,
-        'target=s'          => \my $target,
-        'archive-name=s'    => \my $archive_name,
-        'changes-file=s'    => \my $changes_file,
-        'extra-files=s'     => \my $extra_files,
+        'executable-name=s'     => \my $executable_name,
+        'target=s'              => \my $target,
+        'archive-name=s'        => \my $archive_name,
+        'changes-file=s'        => \my $changes_file,
+        'extra-files=s'         => \my $extra_files,
+        'working-directory=s'   => \my $working_dir,
     );
 
+    if ( $working_dir ) {
+        chdir($working_dir) or die 'Failed to change directory to "' . $working_dir . '" $!.';
```

Comment:
```suggestion
        chdir $working_dir;
```

There's no need for `or die` because of the use of `autodie` in this script.